airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -96,13 +79,102 @@ def get_doc_local_file_path(metadata: ConnectorMetadataDefinitionV0, docs_path:
     return None
 
 
-def compute_gcs_md5(file_name: str) -> str:
-    hash_md5 = hashlib.md5()
-    with open(file_name, "rb") as f:
-        for chunk in iter(lambda: f.read(4096), b""):
-            hash_md5.update(chunk)
+def maybe_create_components_zip(working_directory: Path) -> ManifestOnlyFilePaths:
+    """Create a zip file for components if they exist and return its SHA256 hash."""
+    yaml_manifest_file_path = working_directory / MANIFEST_FILE_NAME
+    components_py_file_path = working_directory / COMPONENTS_PY_FILE_NAME
+
+    if not components_py_file_path.exists():
+        return ManifestOnlyFilePaths(
+            zip_file_path=None,
+            sha256_file_path=None,
+            sha256=None,
+            manifest_file_path=yaml_manifest_file_path,
+        )
+
+    with (
+        tempfile.NamedTemporaryFile(mode="wb", suffix=".zip", delete=False) as zip_tmp_file,
+        tempfile.NamedTemporaryFile(mode="w", suffix=".sha256", delete=False) as sha256_tmp_file,
+    ):
+        python_components_zip_file_path = Path(zip_tmp_file.name)
+        python_components_zip_sha256_file_path = Path(sha256_tmp_file.name)
+
+        files_to_zip: List[Path] = [components_py_file_path, yaml_manifest_file_path]
+        components_zip_sha256 = create_zip_and_get_sha256(files_to_zip, python_components_zip_file_path)
+        sha256_tmp_file.write(components_zip_sha256)
+
+        return ManifestOnlyFilePaths(
+            zip_file_path=python_components_zip_file_path,
+            sha256_file_path=python_components_zip_sha256_file_path,
+            sha256=components_zip_sha256,
+            manifest_file_path=yaml_manifest_file_path,
+        )
+
+
+def _write_metadata_to_tmp_file(metadata_dict: dict) -> Path:
+    """Write the metadata to a temporary file."""
+    with tempfile.NamedTemporaryFile(mode="w", suffix=".yaml", delete=False) as tmp_file:
+        yaml.dump(metadata_dict, tmp_file)
+        return Path(tmp_file.name)
+
+
+# 🛠️ HELPERS
+
+
+def _safe_load_metadata_file(metadata_file_path: Path) -> dict:
+    try:
+        metadata = yaml.safe_load(metadata_file_path.read_text())
+        if metadata is None or not isinstance(metadata, dict):
+            raise ValueError(f"Validation error: Metadata file {metadata_file_path} is invalid yaml.")
+        return metadata
+    except Exception as e:
+        raise ValueError(f"Validation error: Metadata file {metadata_file_path} is invalid yaml: {e}")
+
 
-    return base64.b64encode(hash_md5.digest()).decode("utf8")
+def _any_uploaded(uploaded_files: List[UploadedFile], keys: List[str]) -> bool:
+    """Check if any of the uploaded files have been uploaded."""
```

Comment:
```suggestion
    """Check if the list of uploaded files contains any of the provided keys."""
```
This one broke my brain a little the first time I read it 😆